build.sbt

```diff
@@ -1118,6 +1118,12 @@ val truffleRunOptions = if (java.lang.Boolean.getBoolean("bench.compileOnly")) {
     "-Dpolyglot.engine.BackgroundCompilation=false"
   )
 }
+val truffleRunOptions = "-ea" +: truffleRunOptionsNoAssert
```

Comment:
I am strongly for enabling assertions on as many places as possible, including local dev environments. The sooner the developer sees an error, the better. The only exception to this rule are benchmarks - both local and on CI, and releases.  This can possibly be tweaked also somehow in `build.sbt`.

Reply:
When working on some fundamental changes that might be problematic. But I guess we can live with such an inconvenience.

Reply:
Though if we test _everything_ with `-ea` (I agree that's a good idea) _but_ do releases without `-ea`, then I think we _must_ also have a test suite run (possibly as a release pre-check) without `-ea` too.

Of course, _theoretically_ all `assert`s must not have any observable side effects. But that's the theory. I can easily imagine how disabling an assert may break some code by not running some side effect. Of course such assert would be awfully bad to begin with, but in some subtle cases we may not notice, so we definitely need a check for that - we must have a pre-release check that is consistent with release mode if we are working in a different mode day-to-day, otherwise we will break some release and figure it out too late at some point.

Reply:
Unit tests are run with `-ea`. StdLib tests are probably still executed without `-ea`.

I don't see any need to run (unit) tests with `-ea` and without `-ea`. Nobody is doing that (everybody I know is running tests with `-ea`). Unless it is a blocker for approval, I am not going to investigate how to run tests without `-ea` in CI as part of this PR.

Reply:
Definitely not something to do in this PR I think.

If stdlib tests are run without `-ea` it all sounds good.

Ideally, it would be cool to run stdlib with and without `-ea` on the CI (probably one run on the CI gate and another one nightly, like the benchmarks). But that's something for the future.

It's awesome that we are running at least unit tests and are ensuring the ability to run Enso with `-ea` in our dev builds.